test/functional/system/rounds.js

```diff
@@ -0,0 +1,1584 @@
+											}
+											return balance;
+										},
+										'0'
+									);
+
+									expect(genesisAccount.balance).to.be.equal(balance);
+									expect(genesisAccount.u_balance).to.be.equal(balance);
+									expect(genesisAccount.publicKey.toString('hex')).to.equal(
+										genesisAccountTransaction.senderPublicKey
+									);
+									done();
+								});
+							});
+
+							describe('all accounts', () => {
+								it('balances should be valid against blockchain balances', () => {
+									// Perform validation of accounts balances against blockchain
+									return expect(
+										Queries.validateAccountsBalances()
+									).to.eventually.be.an('array').that.is.empty;
+								});
+							});
+						});
+					});
+				});
+			});
+		});
+
+		describe('modules.delegates', () => {
+			describe('__private.delegatesList', () => {
+				let delegatesList;
+
+				before(() => {
+					return generateDelegateListPromise(1, null).then(_delegatesList => {
+						delegatesList = _delegatesList;
+					});
+				});
+
+				it('should be an array', () => {
+					return expect(delegatesList).to.be.an('array');
+				});
+
+				it('should have a length of 101', () => {
+					return expect(delegatesList.length).to.equal(101);
+				});
+
+				it('should contain public keys of all 101 genesis delegates', done => {
+					_.each(delegatesList, pk => {
+						// Search for that pk in genesis block
+						var found = _.find(library.genesisblock.block.transactions, {
+							senderPublicKey: pk,
+						});
+						expect(found).to.be.an('object');
+					});
+					done();
+				});
+
+				it('should be equal to one generated with Lisk-Core 0.9.3', () => {
+					return expect(delegatesList).to.deep.equal(
+						roundsFixtures.expectedDelegatesOrder
+					);
+				});
+			});
+
+			describe('__private.loadDelegates', () => {
+				before(done => {
+					const loadDelegates = library.rewiredModules.delegates.__get__(
+						'__private.loadDelegates'
+					);
+					loadDelegates(err => {
+						keypairs = library.rewiredModules.delegates.__get__(
+							'__private.keypairs'
+						);
+						done(err);
+					});
+				});
+
+				describe('__private.keypairs', () => {
+					it('should not be empty', () => {
+						expect(keypairs).to.be.an('object');
+						return expect(Object.keys(keypairs).length).to.be.above(0);
+					});
+
+					it('length should match delegates length from config file', () => {
+						return expect(Object.keys(keypairs).length).to.equal(
+							__testContext.config.forging.secret.length
+						);
+					});
+
+					it('every keypairs property should match contained object public key', done => {
+						_.each(keypairs, (keypair, pk) => {
+							expect(keypair.publicKey).to.be.instanceOf(Buffer);
+							expect(keypair.privateKey).to.be.instanceOf(Buffer);
+							expect(pk).to.equal(keypair.publicKey.toString('hex'));
+						});
+						done();
+					});
+				});
+			});
+		});
+	});
+
+	describe('round 1', () => {
```

Comment:
As per my understanding, the round 1 is always with one block which is genesis block. Shouldn't we rename it as `round 2`

Reply:
No, round `1` is applied twice - once (as exception) after genesis block and again after block `101` (`ceil(height/activeDelegates) = round -> ceil(101/101) = 1`).